imagepipeline-base/src/main/java/com/facebook/imageformat/ImageFormatCheckerUtils.java

```diff
@@ -59,5 +59,35 @@ public static boolean startsWithPattern(
     return true;
   }
 
+  /**
+   * Checks if byteArray interpreted as sequence of bytes contains the pattern.
+   * @param byteArray the byte array to be checked
+   * @param pattern the pattern to check
+   * @return index of beginning of pattern, if found; otherwise -1
+   */
+  public static int indexOfPattern(
+      final byte[] byteArray,
+      final byte[] pattern) {
+    Preconditions.checkNotNull(byteArray);
+    Preconditions.checkNotNull(pattern);
+    if (pattern.length > byteArray.length) {
+      return -1;
+    }
+
+    for (int i = 0; i < (byteArray.length - pattern.length); i++) {
+      if (byteArray[i] == pattern[0]) {
+        for (int j = 1; j < pattern.length; j++) {
+          if (byteArray[i + j] != pattern[j]) {
+            break;
+          }
+          if (j == pattern.length - 1) {
```

Comment:
I don't think you should do this check in every loop iteration. Maybe you can do something similar as `String.indexOf(...)` here: http://hg.openjdk.java.net/jdk7u/jdk7u2/jdk/file/58ad18490a50/src/share/classes/java/lang/String.java#l1770